### ISSUE TITLE
ES2406-ca98e75f - Improper Neutralization of ANSI Codes ('ANSI Injection')

### ISSUE THREAD

**Issue by stevechristeycoley** [CWE-CAPEC/CWE-Content-Development-Repository#112]
Submission File: [ES2406-ca98e75f-new-improper-neutralization-ansi-codes-ansi-injection.txt](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/submissions/ES2406-ca98e75f-new-improper-neutralization-ansi-codes-ansi-injection.txt)

ID: ES2406-ca98e75f

SUBMISSION DATE: 2024-06-15 16:03:17

NAME: Improper Neutralization of ANSI Codes ('ANSI Injection')

DESCRIPTION:

With interactive command line interfaces becoming more commonplace with
growing system administration needs and human-friendly TUIs, logging
user-generated values into the terminal into one of these sessions is no
longer safe. ANSI codes that can not only change the color of console
output but can also arbitrarily move the cursor, clear the screen, and make
fake prompts present a danger to the integrity of prompts delivered by the
code inside the interactive CLI and by malicious user input. With a max
score of 8.8/10 on the CVSS ranking, its main application comes from
interactive permission prompts. For example, (CVE-2024-27936) if malicious
user input is added inline inside a permission prompt (i.e the interactive
prompt asks "Can this program run command <user input>," the user input,
with the usage of ANSI codes, can clear the prompt behind it and make a
misleading prompt that asks for an inconspicuous permission. This is a
subcategory of CWE-150, but there are enough CVEs present in this
subcategory (CVE-2003-0020, CVE-2003-0083, CVE-2003-0021, CVE-2003-0022,
CVE-2003-0023, CVE-2003-0063, CVE-2000-0476) to justify this category.

**Comment by cmullaly-mitre** [comment]
This submission has completed the INITIAL REVIEW phase by a CWE Content Team member, who reviewed the submission based on criteria that are intended to ensure that new CWE content meets minimum quality expectations and falls within the current scope of CWE, which simplifies their integration into CWE. (As of January 2024, these criteria are still under development).

- [More details on submission problems](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md)
- [More details on scope exclusions](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/scope-exclusions.md)

Please use ID ES2406-ca98e75f in future correspondence regarding this request and copy cwe-submissions@mitre.org on any inquiry or reply.

This submission contains one or more issues that will require some changes before the submission can be accepted, so it has been moved to the Initial Consultation phase.

The Initial Review has found the following issues, which are summarized. "Resolved" issues (if any) were considered during review but were already addressed, so they require no action.  Detailed explanations are included in the full text of your submission; see "ACTIVE ISSUES - DETAILS" and "RESOLVED ISSUES - DETAILS" sections.

ACTIVE ISSUES:

* SUB.ABS.LOWLEVEL "Low level of abstraction"
* SUB.ACTCHANGE "Change in action for submission"


RESOLVED ISSUES:

None.



Underneath the boundary below, please read the "Comments:" for each issue, enter your "Response:" to each comment, and suggest changes to the submission text. CWE Team members will make the appropriate updates to the raw submission file.

=================================================================


## SUB.ABS.LOWLEVEL - "Low level of abstraction"

URL: [SUB.ABS.LOWLEVEL](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md#SUB.ABS.LOWLEVEL)

*Description*: The submission identifies an issue, but its abstraction might
be too low-level to merit its own CWE entry. If it is focused on a
particular technology, framework, protocol, or function, then it might not
be unique enough to get its own CWE ID.  This might lead to modification of
an existing entry to include the additional details, or creation of a
higher-level entry for which the submission is one example. (Note: some
older CWE entries have this problem and would not be published today,
generally because they were too specific to a particular language.)

*Resolution*: The submission will be allowed to progress to later phases once
the more-general concept is appropriately identified and described. The
original, lower-level submission will serve as a primary example of the
more general concept.

*Comments*: The submission specifically focuses on ANSI codes. CWE-150,
already covers Improper Neutralization of Escape, Meta, or Control Sequences. 
ASNI codes would be classified as "Control Sequences" which is why 
it is covered by CWE-150. Given that CWE-150 is already a variant 
abstraction, the proposed submission is likely to be too low level 
for a new entry. It may be better suited as a modification of CWE-150. 
Please also note that your suggestions of CVE-2003-0020, 
CVE-2003-0083, CVE-2003-0021, and some other observed examples are already 
listed in CWE-150.

*Response*: \<SUBMITTER SHOULD REPLY HERE\><br><br>




## SUB.ACTCHANGE - "Change in action for submission"

URL: [SUB.ACTCHANGE](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md#SUB.ACTCHANGE)

*Description*: The submission's action needs to change, e.g., from "New
Entry" to "Modification," or vice versa. This potentially requires changes
to the kinds of information that will need to be provided.

*Resolution*: The submission cannot progress to Stage 3 (Content-Generation)
until the informational needs are addressed.

*Comments*: Because this submission occurs at such a low level
(SUB.ABS.LOWLEVEL), it might be better to treat this submission as a
modification to other entries (e.g. CWE-150) instead of a separate
new entry. Further discussion and identification of the core weakness
will help determine the appropriate action. Please indicate if you agree
or disagree with this comment.

*Response*: \<SUBMITTER SHOULD REPLY HERE\>

**Comment by tristan-f-r** [comment]
I agree with that judgement, but I don't think I would be able to carry this to a CWE modification.

**Comment by stevechristeycoley** [comment]
The following CWE entry has been modified based on your submission. This CWE will be updated in the next CWE version:

   CWE-150

## CREDITS/ACKNOWLEDGMENTS FOR YOUR CONTRIBUTION:

We plan to credit you using your original contact and organization information. See your email notification for specifics.

* If you wish to change this information, please let us know.

* If you wish to remain anonymous, we will credit an "Anonymous External
  Contributor".

* If we do not receive a response from you, then we will
  use your GitHub username (if any).

The final submission data is included below.

Please let us know if you have any questions.

Thank you for supporting CWE!

The CWE Team

=================================================================

SUBMISSION COMMUNICATION ID: ES2406-ca98e75f<br><br>ACTION TYPE: Modification<br><br>SUBMISSION STATUS: CWE-Modified<br><br>CDRAPPROVED: yes<br><br>AFFECTED CWES: CWE-150<br><br>ORIGIN: sub-server<br><br>SUBMISSION DATE: 2024-06-15 16:03:17<br><br>ISSUES: yes/resolved<br><br>ISSUE DETAILS: <br><br>PUBTRACKER: \#112<br><br>MTRACKER: \#1097<br><br>GITHUBUSER: tristan-f-r<br><br>SUBMISSION TYPE: Software<br><br>NAME: Improper Neutralization of ANSI Codes ('ANSI Injection')<br><br>DESCRIPTION:<br><br>With interactive command line interfaces becoming more commonplace with
growing system administration needs and human-friendly TUIs, logging
user-generated values into the terminal into one of these sessions is no
longer safe. ANSI codes that can not only change the color of console
output but can also arbitrarily move the cursor, clear the screen, and make
fake prompts present a danger to the integrity of prompts delivered by the
code inside the interactive CLI and by malicious user input. With a max
score of 8.8/10 on the CVSS ranking, its main application comes from
interactive permission prompts. For example, (CVE-2024-27936) if malicious
user input is added inline inside a permission prompt (i.e the interactive
prompt asks "Can this program run command \<user input\>," the user input,
with the usage of ANSI codes, can clear the prompt behind it and make a
misleading prompt that asks for an inconspicuous permission. This is a
subcategory of CWE-150, but there are enough CVEs present in this
subcategory (CVE-2003-0020, CVE-2003-0083, CVE-2003-0021, CVE-2003-0022,
CVE-2003-0023, CVE-2003-0063, CVE-2000-0476) to justify this category.<br><br>RELATED WEAKNESSES:<br><br>   ChildOf CWE-150
   ChildOf CWE-74
   Other ES2508-95a780e3<br><br>REFERENCES:<br><br>Title: "u001b[31m"?! ANSI Terminal security in 2023 and finding 10 CVEs
URL: https://dgl.cx/2023/09/ansi-terminal-security<br><br>

**Comment by stevechristeycoley** [comment]
This suggested modification to CWE has been published in the latest CWE version.

This can be seen on the CWE web site at:

     https://cwe.mitre.org/data/definitions/150.html

Credits are provided in the "Contributions" part of the "Content History" at the bottom of the page.

If you see any errors, please let us know.

If there are no problems, we plan to close this submission's issue in CDR after approximately one week.

For transparency and tracking, the original submission will remain archived in CDR under the closed/ directory.

Thank you for supporting CWE!

The CWE Team

**Comment by cmullaly-mitre** [comment]
This submission has been fully processed and will now be closed.  No
further action is needed on your part.

Thank you for supporting CWE!

The CWE Team
